### ISSUE TITLE
Events are delivered twice to "on event from" handler with value MICROBIT_EVT_ANY

### ISSUE THREAD

**Issue by martinwork** [microsoft/pxt-microbit#2382]
**Describe the bug**

When using an "on event from" handler with value MICROBIT_EVT_ANY, each event is delivered twice. 

Tested using Button A events and MES_DPAD events. 

Occurs on live and beta but not on v1.

In addition, the simulator shows the wrong events for button A presses..

**To Reproduce**
1. Go to 'https://makecode.microbit.org/_EFP7X50XiK78'
2. Download to micro:bit
3. Press button A down
4. See Heart-1, Heart-1 repeated
5. Change "with value" to MICROBIT_BUTTON_EVT_DOWN
2. Download to micro:bit
3. Press button A down
4. See Heart-1 just once

**Expected behavior**
Each event should be delivered just once each time it is sent.

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
Live
makecode.microbit.org version:  2.0.6
Microsoft MakeCode version:  5.15.12
microbit runtime version:  pxtgc-0

And beta
makecode.microbit.org version:  2.1.4
Microsoft MakeCode version:  5.19.8
microbit runtime version:  pxtgc-0

**Comment by philipphenkel** [comment]
This could be regression caused by the fix for #1334. When fixing this one we should double-check the other test case as well.

Btw, I tested on a device and see a different (wrong) behavior in step 4:
See Heart-1, Heart-1 repeated, See Heart-1, Heart-2 repeated, See Heart-3, Heart-3 repeated

**Comment by martinwork** [comment]
Thanks for confirming @philipphenkel . That is what I see when I press and release, but I left out the details! A quick press-release will be 1,2,3 (down,up,click) or a long press and release will be 1,5,2 (down,hold,up). Each of these events is delivered twice for MICROBIT_EVT_ANY but only event 1 is comparable with the MICROBIT_BUTTON_EVT_DOWN case. I think it's any event too, not just the buttons. I was trying to use the gamepad events.

**Comment by philipphenkel** [comment]
This could be a duplicate of #2255 (see https://github.com/microsoft/pxt-microbit/issues/2255#issuecomment-529097552 for a generic test example)

**Comment by martinwork** [comment]
Yes @philipphenkel . This looks like the same or very closely related problem.

**Comment by martinwork** [comment]
I think the problem lies in incompatibilities in various versions of findBinding and dispatchEvent.

See https://github.com/microsoft/pxt-common-packages/pull/829/commits/1167fa3e0fb5190ed444f58b91dc6aea46735921

The action of findBinding has been changed to return the ANY binding as well as the requested one, with the addition of a nextBinding function and insertion of the old findBinding implementation into  setBinding.

The implementations of dispatchEvent in pxt-common-packages have been changed to use the new findBinding  and nextBinding, but the implementation of dispatchEvent in pxt-microbit assumes the old findBinding behaviour. It tries to find the specific binding then the ANY binding, but will find the ANY binding twice unless the specific one is first in the list (defined last).

**Comment by abchatra** [comment]
Old bug. Not planning to address.
